RFCs/RFC-001: Introspect streams.md

```diff
@@ -0,0 +1,62 @@
+# RFC-001: Introspect Streams
+
+**To be Reviewed By:** 11/19/2019
+
+**Authors:** Swapnil Bawaskar
+
+**Status:** **Active** | Accepted | Dropped | Superseded
+
+**Superseded by:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When the streaming runtime is installed, riff users can create streams (backed by a stream provider e.g. kafka). Users can write functions that read from and write to these streams,
+but there is no easy mechanism to get events into the stream and look into the contents of the stream which are helpful while developing/demoing.
+
+### Anti-Goals
+We will only address this problem for development/demos, not production, so topics like auth/authz are out of scope for this document.
+
+## Solution
+### Ideal solution:
+The  is to provide users with only one cli (riff cli) that will enable users to send/receive messages to/from a stream (commands below). We will have to ask the users to port-forward the gateway service if they are running outside the cluster.
```

Comment:
adding commands to the `riff` cli seems out of alignment with the stated anti-goal of not being appropriate for production.

Reply:
I believe that we should not ask the user to download two CLIs. We can mark the individual commands as "experimental".

Reply:
There's a difference between something that is currently an experiment that expect to solidify, and something that is not intended for production uses.

Based on the problem statement, I continue to assert that the riff cli is not an appropriate home for this functionaltiy. It will be much easier down the road to add the functionality into the riff cli, then remove it from the cli.

Reply:
Would it then suffice to remove the "not for production" from the anti-goal?

Reply:
I also think we should keep the riff CLI focused only on production usage (same reason I argued for removal of the "invoke" command).

I don't see any problem with separate (go-binary or shell) *commands* - not necessarily full "CLIs" themselves - for things like invoke, publish, and subscribe. We could also create an image that includes those commands in a single directory, so that it's easy to run within the cluster, e.g. `kubectl exec ... -- publish --payload ...`